src/main/java/org/spongepowered/api/text/Text.java

```diff
@@ -107,6 +130,8 @@
         this.clickAction = Optional.<ClickAction<?>>ofNullable(clickAction);
         this.hoverAction = Optional.<HoverAction<?>>ofNullable(hoverAction);
         this.shiftClickAction = Optional.<ShiftClickAction<?>>ofNullable(shiftClickAction);
+        this.childrenIterable = () ->
+                this.children.isEmpty() ? Iterators.singletonIterator(this) : new TextIterator(this);
```

Comment:
You can test it, it will fail to compile in some cases (for whatever reason), it complains that children might not be initialized yet.